### PR TITLE
Input

### PR DESCRIPTION
**Pull request by cookiester34**
Restructured input handling, no longer in boco_window package, moved events to boco-events and input handling to boco_input

### CODE REVIEW

File: src/boco_core/boco_input/Input_handling.odin [hunk 1]
```diff
@@ -0,0 +1,49 @@
+package boco_input
+
+import "core:log"
+import "../boco_event_data"
+
+Input_event :: struct{
```
Comment (cookiester34) [inline src/boco_core/boco_input/Input_handling.odin:6]: Was thinking about turning this into a union but not sure

Comment (Benasm123) [reply src/boco_core/boco_input/Input_handling.odin:6]: If the Input_event can never be both a key_event and a mouse_event then yeah union all the way.

File: src/boco_core/boco_input/Input_handling.odin
Comment (Benasm123) [inline src/boco_core/boco_input/Input_handling.odin:1]: Inconsistent naming, using upper case first letter here but not i other files

**Review by Benasm123 — APPROVED** [review]

File: src/boco_core/boco_input/Input_handling.odin
Comment (cookiester34) [reply src/boco_core/boco_input/Input_handling.odin:1]: I thought all structs start upper case

**Pull request merged**